torch/csrc/autograd/functions/jit_closure.cpp

```diff
@@ -565,17 +567,14 @@ struct StageClosure {
     auto graph = node->owningGraph();
     // Gather uses of each output
     std::vector<std::reference_wrapper<const use_list>> output_uses_refs;
-    if (node->hasMultipleOutputs()) {
-      // Each use is a single Select node corresponding to an output
-      for (auto& use : node->uses()) {
-        if (use.user->isHandle()) continue;
-        auto& select_uses = use.user->uses();
-        output_uses_refs.emplace_back(select_uses);
-      }
-    } else {
-      output_uses_refs.emplace_back(node->uses());
+    for (auto o : node->outputs()) {
+      if (o->isHandle()) continue;
+      auto& select_uses = o->uses();
+      output_uses_refs.emplace_back(select_uses);
     }
-
+    fillNextFunctions(graph, output_uses_refs, fn, node_map, output_offset, stage);
+  }
+  void fillNextFunctions(Graph * graph, std::vector<std::reference_wrapper<const use_list>> & output_uses_refs, const std::shared_ptr<Function>& fn, node_fn_map_type& node_map, int output_offset, std::size_t stage) {
```

Comment:
omg what even happened to the types in here

Reply:
Yeah, since this code is going away soon I didn't try very hard and just refactored mechanically so that Input nodes would work.